tests/sessions_tests/tests.py

```diff
@@ -519,19 +519,25 @@ def test_non_default_cache(self):
         with self.assertRaises(InvalidCacheBackendError):
             self.backend()
 
+    @override_settings(
+        CACHES={"default": {"BACKEND": "sessions_tests.cache.FailingCache"}}
+    )
     def test_session_cache_failure(self):
-        max_value_length = 2**20
+        self.session = self.backend()
 
-        exceptions = []
-        key = "some key"
-        val = "a" * (max_value_length + 1)
-        try:
-            self.session[key] = val
+        with self.assertLogs("django.request", "ERROR") as cm:
             self.session.save()
-        except Exception as e:
-            exceptions.append(e)
 
-        self.assertEqual(len(exceptions), 0)
+        error_records = [record for record in cm.records if record.levelname == "ERROR"]
+
+        self.assertTrue(error_records)
+        self.assertTrue(
+            error_records[0].getMessage().startswith("Error saving to cache")
+        )
+
+        self.session["key"] = "val"
+        self.session.save()
+        self.assertEqual(self.session["key"], "val")
```

Comment:
There's no need for such complexity in this test (ie the list comp, 2 assert trues)... it's better to keep it simple:

```
        self.session = self.backend()
        self.session["key"] = "val"
        with self.assertLogs("django.request", "ERROR") as cm:
            self.session.save()
        self.assertEqual(
            cm.records[0].message, "Error saving to cache: Error saving to cache"
        )
        self.assertEqual(self.session["key"], "val")
```